base/interactiveutil.jl

```diff
@@ -254,8 +254,13 @@ function gen_call_with_extracted_types(fcn, ex0)
                     Expr(:call, typesof, map(esc, args[2:end])...))
     end
     exret = Expr(:none)
+    is_macro = false
     ex = expand(ex0)
-    if !isa(ex, Expr)
+    if fcn in [:which, :less, :edit, :functionloc] && ex0.head == :macrocall
```

Comment:
I see where you are going now:). I have removed the special casing of `@edit` etc. The Ast displayed by `@code_warntype` is pretty ugly, but it displays.
```
julia> @code_warntype @time 1+1
Variables:
  #self#::Base.#@time
  ex::Expr
  ####args#787#8911::TUPLE

Body:
  begin  # util.jl, line 152:
      return (top(_expr))(:block,$(Expr(:copyast, :($(QuoteNode(:( # util.jl, line 153:)))))),$(Expr(:copyast, :($(QuoteNode(:(local stats = gc_num())))))),$(Expr(:copyast, :($(QuoteNode(:( # util.jl, line 154:)))))),$(Expr(:copyast, :($(QuoteNode(:(local elapsedtime = time_ns())))))),$(Expr(:copyast, :($(QuoteNode(:( # util.jl, line 155:)))))),(top(_expr))(:local,(top(_expr))(:(=),:val,(Core._expr)(:escape,ex::Expr)::ANY)::Expr)::Expr,$(Expr(:copyast, :($(QuoteNode(:( # util.jl, line 156:)))))),$(Expr(:copyast, :($(QuoteNode(:(elapsedtime = time_ns() - elapsedtime)))))),$(Expr(:copyast, :($(QuoteNode(:( # util.jl, line 157:)))))),$(Expr(:copyast, :($(QuoteNode(:(local diff = GC_Diff(gc_num(),stats))))))),$(Expr(:copyast, :($(QuoteNode(:( # util.jl, line 158:)))))),$(Expr(:copyast, :($(QuoteNode(:(time_print(elapsedtime,diff.allocd,diff.total_time,gc_alloc_count(diff)))))))),$(Expr(:copyast, :($(QuoteNode(:( # util.jl, line 160:)))))),:val)::Expr
  end::Expr
```